### PR TITLE
[CI] Add test package creating local resources with Terraform

### LINKED ISSUE #1952: Replace test packages creating resources in AWS (terraform) by packages creating resources locally
As part of the testing performed by `elastic-package` there are packages that create resources using Terraform in order to test the corresponding service deployer ([Terraform service deployer docs](https://github.com/elastic/elastic-package/blob/23aaa908845814df67789be9f407cc4c76669c89/docs/howto/system_testing.md#terraform-service-deployer))

Currently, as part of these test packages, there are two packages creating resources in cloud providers (AWS):
- [`aws` package](https://github.com/elastic/elastic-package/tree/23aaa908845814df67789be9f407cc4c76669c89/test/packages/parallel/aws)
- [`aws_logs` package](https://github.com/elastic/elastic-package/tree/23aaa908845814df67789be9f407cc4c76669c89/test/packages/parallel/aws_logs)

In order to avoid issues with limits/quotas or even network issues it would be interesting to replace those two packages by a test package that create resources locally using Terraform. This also would allow us to remove the usage of these credentials in CI.

Considerations when replacing those two packages:
- `aws` is a package that also contains multi-tiles in their manifests.
- `aws_logs` tests the use of Terraform outputs. https://github.com/elastic/elastic-package/issues/703

As a note, previously there was a package creating resources in GCP too (with just 1 data stream) in case it is needed to add some other consideration: https://github.com/elastic/elastic-package/pull/1930

### PR DESCRIPTION
**Pull request by mrodm**
Resolves #1952 

This PR adds a new test package that uses Terraform to create local resources (files). Those files will be read by the Elastic Agent (as logs) via the volume mount `SERVICE_LOGS_DIR`.

This test package includes terraform resources and outputs as they are used in `aws` or `aws_logs` packages.

For instance in aws_logs package is used outputs as:
- https://github.com/elastic/elastic-package/blob/48ad94bd16cb08416b9ba6d30b71b716656faecc/test/packages/parallel/aws_logs/data_stream/generic/_dev/test/system/test-default-config.yml#L11
- https://github.com/elastic/elastic-package/blob/48ad94bd16cb08416b9ba6d30b71b716656faecc/test/packages/parallel/aws_logs/data_stream/generic/_dev/deploy/tf/main.tf#L63

Example using outputs from Terraform to define variables:
variables using outputs

In this case, the file path is dynamic and the value is set in the terraform run (using `TF_RUN_ID` variable). From the Terraform output:

```
 Terraform will perform the following actions:
 
   # local_file.log will be created
   + resource "local_file" "log" {
       + content_base64sha256 = (known after apply)
       + content_base64sha512 = (known after apply)
       + content_md5          = (known after apply)
       + content_sha1         = (known after apply)
       + content_sha256       = (known after apply)
       + content_sha512       = (known after apply)
       + directory_permission = "0777"
       + file_permission      = "0777"
       + filename             = "/tmp/service_logs/90231.log"
       + id                   = (known after apply)
       + source               = "./files/example.log"
     }
 
 Plan: 1 to add, 0 to change, 0 to destroy.
 
 Changes to Outputs:
   + filename = "/tmp/service_logs/90231.log"

``` 

## How to test locally

```shell
cd test/packages/parallel/terraform_local
elastic-package stack up -v -d

# both tests should ingest one document in Elasticsearch
elastic-package test system -v --data-streams local
elastic-package test system -v --data-streams outputs
elastic-package test system -v

elastic-package stack down -v
```

## Author checklist
- [x] Remove aws_logs package
- [x] Remove system tests of aws package. Keep testing multi-tile installation.

### CODE REVIEW

File: test/packages/parallel/terraform_local/data_stream/outputs/_dev/deploy/tf/main.tf [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+}
+
+resource "local_file" "log" {
+  source          = "./files/example.log"
+  filename        = "/tmp/service_logs/${var.TEST_RUN_ID}.log"
+  file_permission = "0777"
+}
+
+output "filename" {
+    value = local_file.log.filename
```
Comment (mrodm) [inline test/packages/parallel/terraform_local/data_stream/outputs/_dev/deploy/tf/main.tf:12]: In this data stream , a Terraform output is set and it is used to configure the policy in the system configuration file.

File: test/packages/parallel/terraform_local/data_stream/local/_dev/deploy/tf/main.tf [hunk 2]
```diff
@@ -0,0 +1,5 @@
+resource "local_file" "log" {
+  source          = "./files/example.log"
+  filename        = "/tmp/service_logs/file.log"
```
Comment (mrodm) [inline test/packages/parallel/terraform_local/data_stream/local/_dev/deploy/tf/main.tf:3]: This folder is also accessible by the Elastic Agent, so this file created by Terraform will be used to ingest logs into Elasticsearch.

File: internal/servicedeployer/_static/terraform_deployer.yml [hunk 3]
```diff
@@ -13,3 +13,4 @@ services:
     volumes:
       - ${TF_DIR}:/stage
       - ${TF_OUTPUT_DIR}:/output
+      - ${SERVICE_LOGS_DIR}:/tmp/service_logs/
```
Comment (mrodm) [inline internal/servicedeployer/_static/terraform_deployer.yml:16]: Required to create files that could be accessible by the Elastic Agents.

This folder/volume is also used by the services (e.g. nginx):
https://github.com/elastic/elastic-package/blob/48ad94bd16cb08416b9ba6d30b71b716656faecc/test/packages/parallel/nginx/_dev/deploy/docker/docker-compose.yml#L8

**Comment by mrodm** [comment]
As this new test package is testing that terraform works creating resources locally and also checks that can be used terraform outputs as part of variables:
- `aws_logs` test package was removed.
- system tests for `aws` test package were removed.

Do you think it should be added anything else to the `terraform_local` test package ? @elastic/ecosystem

**Review by jsoriano — APPROVED** [review]
Thanks!

**Pull request merged**